src/d3d9/d3d9_device.cpp

```diff
@@ -3777,13 +3777,19 @@ namespace dxvk {
         m_dirtySamplerStates |= 1u << StateSampler;
       }
 
-      if (unlikely(m_fetch4Enabled & (1u << StateSampler) && !(m_fetch4 & (1u << StateSampler)))) {
+      if (unlikely(m_fetch4Enabled & (1u << StateSampler))) {
         bool textureSupportsFetch4 = newTexture->SupportsFetch4();
-        if (textureSupportsFetch4
+        if (unlikely(textureSupportsFetch4
           && m_state.samplerStates[StateSampler][D3DSAMP_MAGFILTER] == D3DTEXF_POINT
-          && m_state.samplerStates[StateSampler][D3DSAMP_MINFILTER] == D3DTEXF_POINT) {
+          && m_state.samplerStates[StateSampler][D3DSAMP_MINFILTER] == D3DTEXF_POINT
+          && !(m_fetch4 & (1u << StateSampler)))) {
+          // We swapped a multi channel texture for a single channel texture => turn on fetch4
           m_fetch4 |= 1u << StateSampler;
           m_dirtySamplerStates |= 1u << StateSampler;
+        } else if (unlikely(m_fetch4 & (1u << StateSampler))) {
```

Comment:
I don't think this else if is right? Either way, lets clean up this logic into separate bools, it's getting overly complex to follow at this point.

Reply:
Oh, you're right. I accidently dropped the format check when I added the comments.